server/server.go

```diff
@@ -114,30 +118,82 @@ func (s *Server) Version(ctx context.Context, req *v1alpha1.VersionRequest) (*v1
 // handleMountEvent fetches the secrets from the secretmanager API and
 // writes them to the filesystem based on the SecretProviderClass configuration.
 func handleMountEvent(ctx context.Context, client *secretmanager.Client, cfg *config.MountConfig) ([]*v1alpha1.ObjectVersion, error) {
-	ovs := []*v1alpha1.ObjectVersion{}
-	for _, secret := range cfg.Secrets {
-		req := &secretmanagerpb.AccessSecretVersionRequest{
-			Name: secret.ResourceName,
-		}
-
-		result, err := client.AccessSecretVersion(ctx, req)
-		if err != nil {
-			// TODO: determine error codes, should we propagate error code space
-			// from the secret call to this response?
-			klog.ErrorS(err, "failed to access secret", "secret", secret.ResourceName, "pod", klog.ObjectRef{Namespace: cfg.PodInfo.Namespace, Name: cfg.PodInfo.Name})
-			return nil, status.Error(codes.Internal, err.Error())
-		}
+	results := make([]*secretmanagerpb.AccessSecretVersionResponse, len(cfg.Secrets))
+	errs := make([]error, len(cfg.Secrets))
+
+	// In parallel fetch all secrets needed for the mount
```

Comment:
Is there a backoff in the go client library in case too many parallel requests consume all the quota?

Reply:
Ah retries are configured to have backoff: https://github.com/googleapis/googleapis/blob/e195c9cd5a922cc5252b83b0c42a1f521305a96a/google/cloud/secretmanager/v1/secretmanager_grpc_service_config.json#L72-L80but not for RESOURCE_EXHAUSTED responses. I believe those arn't marked as retry-able to avoid amplifying the problem.

We could limit concurrency here but I don't know that that would have much effect since parallelism will actually be spread out across the number of pod's being started.

Reply:
Thanks - that makes to me. Limiting concurrency here could help us limit total in-flight requests to # pods instead of # pods * # secrets, but seeing as quota is enforced per minute and the system is self-healing (ish) due to the 60s rotation, this seems fine.

Should we document that this is expected if quota errors are hit? (i.e. pods will be restarted a few times according to the restart policy and then fix themselves hopefully).

Reply:
Yeah i think this will come up in #32 (load tests)